vmdb/app/models/container_node.rb

```diff
@@ -3,5 +3,6 @@ class ContainerNode < ActiveRecord::Base
   # :name, :uid, :creation_timestamp, :resource_version
   belongs_to :ext_management_system, :foreign_key => "ems_id"
   has_many   :container_groups
+  has_one    :hardware, :dependent => :destroy
```

Comment:
Yeah ... The container node needs more than just the hardware reference.  It should also depend on an operating system which is encapsulated in the `ComputerSystem`.  I like this idea.